lib/proxy.js

```diff
@@ -94,9 +94,9 @@ function buildStream (host, options, secure) {
 ////////////////////////////////////////////////////////////
 
 function buildResultingMiddleware (host, options, stream) {
-  console.log(stream)
   return function proxyMiddleware (req, res, next) {
 
+    var _stream = _.clone(stream, true);
```

Comment:
I think that would still create the same problem because you are cloning outside of each request. so each request will modify that cloned object creating the same issue as before.

Are you that worried about performance with cloning on each request? Do you have tests showing it being that bad?